service/history/queues/executable.go

```diff
@@ -90,6 +89,9 @@ const (
 	// resourceExhaustedResubmitMaxAttempts is the same as resubmitMaxAttempts but only applies to resource
 	// exhausted error
 	resourceExhaustedResubmitMaxAttempts = 1
+	// taskCriticalAttempts, if exceeded, task attempts metrics and critical processing error log will be emitted
+	// while task is retrying
+	taskCriticalAttempts = 30
```

Comment:
I would call it more explicitly: `skipLogFirstTaskFailureAttempts` or something like this.